modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -34,304 +39,208 @@ extension AddressBookClient: DependencyKey {
 
         return Self(
             allLocalContacts: {
-                // return latest known contacts
+                // return latest known contacts or load ones for the first time
                 guard latestKnownContacts == nil else {
-                    if let contacts = latestKnownContacts {
-                        return contacts
-                    } else {
-                        return .empty
-                    }
+                    return latestKnownContacts ?? .empty
                 }
 
                 // contacts haven't been loaded from the locale storage yet, do it
                 do {
                     guard let documentsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first else {
                         throw AddressBookClientError.documentsFolder
                     }
-                    let fileURL = documentsDirectory.appendingPathComponent(Constants.component)
-                    let encryptedContacts = try Data(contentsOf: fileURL)
 
-                    let decryptedContacts = try AddressBookClient.decryptData(encryptedContacts)
-                    latestKnownContacts = decryptedContacts
+                    // Try to find and get the data from the encrypted file with the latest encryption version
+                    let encryptedFileURL = documentsDirectory.appendingPathComponent(try AddressBookClient.filenameForEncryptedFile())
 
-                    return decryptedContacts
+                    if let contactsData = try? Data(contentsOf: encryptedFileURL) {
+                        // file exists, try to find the unencrypted file in case the delete failed
+                        let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
+                        if FileManager.default.fileExists(atPath: unencryptedFileURL.path) {
+                            try? FileManager.default.removeItem(at: unencryptedFileURL)
+                        }
+                        
+                        let contacts = try AddressBookClient.contactsFrom(encryptedData: contactsData)
+
+                        latestKnownContacts = contacts
+                        return contacts
```

Comment:
Why do we modify `latestKnownContacts` at every return, rather than having the caller set it?

Reply:
Goal here is to get the local contacts but we don't want to read it from the file every time `allLocalContacts` is called. Instead, it's read only for the first time and next time `latestKnownContacts` delivers the contacts from in-memory. The technology behind this is the dependency in immutable so there must be a subject that can do it for us. `latestKnownContacts` is the mutable subject.